### PR TITLE
Add feature to build rules based on users, groups on Proxy/Squid

### PR DESCRIPTION
**Pull request by kekek2**

### CODE REVIEW

**Comment by fichtner** [comment]
yay!

**Comment by AdSchellevis** [comment]
@kekek2 thanks for this, but can you please check you code for PSR-2 compatibility (https://docs.opnsense.org/development/guidelines.html)?  There are quite some missing curly braces in there.

File: www/web-proxy-useracl/Makefile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+PLUGIN_NAME=		web-proxy-useracl
+PLUGIN_VERSION=		1.0.0
```
Comment (fichtner) [inline www/web-proxy-useracl/Makefile]: Can we start with 0.1 ?

This may be core.git material

File: www/web-proxy-useracl/Makefile [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+PLUGIN_COMMENT=		Group & User Squid ACL
+PLUGIN_DEPENDS=		web-proxy-sso
+PLUGIN_MAINTAINER=	kekek2@ya.ru
+PLUGIN_WWW=			http://smart-soft.ru
```
Comment (fichtner) [inline www/web-proxy-useracl/Makefile]: PLUGIN_DEVEL= yes

File: www/web-proxy-useracl/src/opnsense/mvc/app/models/OPNsense/ProxyUserACL/ProxyUserACL.php [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?php
+
```
Comment (fichtner) [inline www/web-proxy-useracl/src/opnsense/mvc/app/models/OPNsense/ProxyUserACL/ProxyUserACL.php]: all files will have to have license headers and your copyright ;)

**Comment by kekek2** [comment]
This plugin depends on https://github.com/opnsense/core/pull/1892

File: www/web-proxy-useracl/Makefile [hunk 4]
```diff
@@ -0,0 +1,9 @@
+PLUGIN_NAME=		web-proxy-useracl
+PLUGIN_VERSION=		0.0.1
+PLUGIN_COMMENT=		Group & User Squid ACL
+PLUGIN_DEPENDS=		os-web-proxy-sso
```
Comment (fichtner) [inline www/web-proxy-useracl/Makefile]: why is that? if you depend on your own plugin, shouldn't this code be in the other plugin as well? :)

File: www/web-proxy-useracl/src/opnsense/mvc/app/models/OPNsense/ProxyUserACL/Menu/Menu.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<menu>
+	<Services>
+		<WebProxy>
+			<ProxyUserACL VisibleName="Group and User ACL" cssClass="fa fa-shield fa-fw" order="30" url="/ui/proxyuseracl/" >
```
Comment (fichtner) [inline www/web-proxy-useracl/src/opnsense/mvc/app/models/OPNsense/ProxyUserACL/Menu/Menu.xml]: this can end with /&gt; omitting the line below

**Comment by kekek2** [comment]
It can use proxy-sso:
https://github.com/kekek2/plugins/blob/ProxyUserACL/www/web-proxy-useracl/src/opnsense/service/templates/OPNsense/ProxyUserACL/ProxyUserACL.conf#L23
But proxy-sso is not required

File: www/web-proxy-useracl/src/opnsense/mvc/app/models/OPNsense/ProxyUserACL/Menu/Menu.xml (hunk 5 above)
Comment (fichtner) [reply www/web-proxy-useracl/src/opnsense/mvc/app/models/OPNsense/ProxyUserACL/Menu/Menu.xml]: fa-shield is used for intrusion detection, please use fa-users

**Comment by fichtner** [comment]
Thanks, I'll do some post-op stuff for style, whitespace and menu

**Pull request merged**